subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/NonHierarchicalFileWatcherUpdater.java

```diff
@@ -89,7 +89,8 @@ public void registerWatchableHierarchy(File watchableHierarchy, SnapshotHierarch
     }
 
     @Override
-    public SnapshotHierarchy buildStarted(SnapshotHierarchy root) {
+    public SnapshotHierarchy buildStarted(SnapshotHierarchy root, WatchMode watchMode) {
+        watchableHierarchies.updateUnwatchableFileSystems(watchMode);
```

Comment:
Since we are doing this for both hierarchical and non-hierarchical file systems, can we move this to `DefaultFileWatcherRegistry.buildStarted()` instead?

Reply:
I created an `AbstractFileWatcherUpdater` which takes care of calling the method. Moving it to `DefaultFileWatcherRegistry` would require to expose another method on `FileWatcherUpdater`.